src/core/lib/iomgr/polling_interface.h

```diff
@@ -0,0 +1,167 @@
+/*
+ *
+ * Copyright 2015 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_CORE_LIB_IOMGR_POLLING_INTERFACE_H
+#define GRPC_CORE_LIB_IOMGR_POLLING_INTERFACE_H
+
+#include <grpc/support/port_platform.h>
+#include <grpc/support/sync.h>
+#include <grpc/support/time.h>
+
+#include "src/core/lib/iomgr/exec_ctx.h"
+
+#ifndef NDEBUG
+extern grpc_tracer_flag grpc_trace_fd_refcount;
+#endif
+
+namespace grpc_core {
+
+// Forward declarations
+class PollingJoin;
+
+// A Pollable is an object that can be polled
+// Polling engines derive from this interface and implement
+// This interface should not be implemented outside of a polling engine
+class Pollable {};
+
+// A PollableCollection is a front-end interface to Poller & PollingJoin,
+// collecting common methods.
+// In many situations we don't care which we're dealing with, and so this class
+// serves to abstract away that detail
+class PollableCollection {
```

Comment:
"Collection" makes me think of a sequence or set of elements. Not that the old "entity" was great, but it didn't convey multiplicity like "collection" IMHO.

Reply:
Names are important here... these are the best I've come up with, but I want consensus.

If something has only one operation (add), is it not a collection or a set?
What's the thing that it should convey that it's not?

Reply:
What about PollableSet?

Reply:
PollableSet SGTM

Reply:
Updated